ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1905,14 +2009,16 @@ IopCreateDeviceInstancePath(
         return Status;
     }
 
+    IsValidID = IopValidateID((PWCHAR)IoStatusBlock.Information, BusQueryDeviceID);
+
+    if (!IsValidID)
+    {
+        DPRINT1("Invalid DeviceID. DeviceNode - %p\n", DeviceNode);
+    }
+
```

Comment:
I suppose you should do more than that, either returning FALSE here, or bugchecking.